.github/workflows/code-checks.yml

```diff
@@ -26,17 +32,20 @@ jobs:
       uses: actions/cache@v3
       with:
         path: ~/.cache/pip
-        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/pytorch/base.txt') }}
+        key: ${{ runner.os }}-pip-${{ matrix.name }}-${{ hashFiles('requirements/pytorch/base.txt') }}
         restore-keys: |
           ${{ runner.os }}-pip-
 
-    - name: Install dependencies
+    - name: Adjust dependencies
+      if: matrix.name == "pytorch"
       run: |
         pip install torch==1.12 --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
         python ./requirements/pytorch/adjust-versions.py requirements/pytorch/extra.txt
-        # todo: adjust requirements for both code-bases
-        pip install -r requirements/pytorch/devel.txt --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
+
+    - name: Install dependencies
+      run: |
+        pip install -r requirements/${{ matrix.name }}/devel.txt --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
         pip list
 
     - name: Check typing
-      run: mypy
+      run: mypy ./src/${{ matrix.pkg }}
```

Comment:
There seems to be an issue in the apps source:

```python
src/lightning_app/cli/component-template/placeholdername/__init__.py: error: Duplicate module named "placeholdername" (also at "src/lightning_app/cli/app-template/placeholdername/__init__.py")
src/lightning_app/cli/component-template/placeholdername/__init__.py: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules for more info
src/lightning_app/cli/component-template/placeholdername/__init__.py: note: Common resolutions include: a) using `--exclude` to avoid checking one of them, b) adding `__init__.py` somewhere, c) using `--explicit-package-bases` or adjusting MYPYPATH
```